src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java

```diff
@@ -121,40 +100,20 @@ public void flush(final Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
 
     private void flushFile(final String filename, final List<SinkRecord> records) {
         final BlobInfo blob = BlobInfo
-            .newBuilder(config.getBucketName(), config.getPrefix() + filename)
-            .build();
-
-        try (final ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
-            // Don't group these two tries,
-            // because the internal one must be closed before writing to GCS.
-            try (final OutputStream compressedStream = getCompressedStream(baos)) {
-                try (final OutputWriter outputWriter = getOutputWriter(compressedStream)) {
-                    for (final SinkRecord record: records) {
-                        outputWriter.writeRecord(record);
-                    }
-                }
-            }
-            storage.create(blob, baos.toByteArray());
+                .newBuilder(config.getBucketName(), config.getPrefix() + filename)
+                .build();
+        try (final var out = Channels.newOutputStream(storage.writer(blob));
+             final var writer = OutputWriter.builder()
+                     .withExternalProperties(config.originalsStrings())
```

Comment:
Sorry, maybe I missed something. What does exactly originalStrings are responsible for?

Reply:
by default config is `Map<String, Object>`, but props are passed as strings, so `originalsStrings` returns `Map<String, String>`